core-library/src/main/java/org/silverpeas/core/admin/domain/DomainDriverManager.java

```diff
@@ -983,196 +746,103 @@ public Domain getDomain(String domainId) throws Exception {
+      throw new AdminException(failureOnGetting("driver of domain", domainId), e);
     }
+    return domainDriver;
   }
 
   /**
-* Commit transaction in specific domain driver
-*/
-  public void commit(String domainId) throws Exception {
-    try {
-      // Get a DomainDriver instance
-      DomainDriver domainDriver = this.getDomainDriver(idAsInt(domainId));
-      // Commit transaction
-      domainDriver.commit();
-    } catch (Exception e) {
-      throw new AdminException("Fail to commit transaction for domain " + domainId, e);
-    }
+   * Called when Admin starts the synchronization on a particular Domain
+   */
+  public void beginSynchronization(String sdomainId) throws AdminException {
+    // Get a DomainDriver instance
+    DomainDriver domainDriver = this.getDomainDriver(sdomainId);
+    domainDriver.beginSynchronization();
   }
 
   /**
-* Rollback transaction in specific domain driver
-*/
-  public void rollback(String domainId) throws Exception {
-    try {
-      // Get a DomainDriver instance
-      DomainDriver domainDriver = this.getDomainDriver(idAsInt(domainId));
-      // Commit transaction
-      domainDriver.rollback();
-    } catch (Exception e) {
-      throw new AdminException("Fail to rollback transaction for domain " + domainId, e);
-    }
+   * Called when Admin ends the synchronization
+   * @param cancelSynchro true if the synchronization is cancelled, false if it ends normally
+   */
+  public String endSynchronization(String sdomainId, boolean cancelSynchro) throws AdminException {
+    // Get a DomainDriver instance
+    DomainDriver domainDriver = this.getDomainDriver(sdomainId);
+    return domainDriver.endSynchronization(cancelSynchro);
   }
 
-  protected String[] translateUserIdsToSpecificIds(int domainId, String[] ids) {
-    List<String> specificIds = new ArrayList<String>();
-    if (ids == null) {
-      return null;
+  private String[] translateUserIdsToSpecificIds(String domainId, String[] ids)
+      throws AdminException {
+    if (ids == null || ids.length == 0) {
+      return ArrayUtil.EMPTY_STRING_ARRAY;
     }
 
-    for (String id : ids) {
-      // Get the user information
-      try {
-        UserRow ur = getOrganization().user.getUser(idAsInt(id));
-        if ((ur != null) && (ur.domainId == domainId)) {
-          specificIds.add(ur.specificId);
-        }
-      } catch (Exception e) {
-        SilverTrace.error("admin", "DomainDriverManager.getUser",
-                "admin.MSG_ERR_GET_USER", "user Id: '" + id + "'", e);
-      }
+    try(Connection connection = DBUtil.openConnection()) {
+      List<UserDetail> users = userDAO.getUsersByCriteria(connection,
+          UserSearchCriteriaForDAO.newCriteria().onDomainId(domainId).onUserIds(ids));
+      List<String> specificIds =
+          users.stream().map(UserDetail::getSpecificId).collect(Collectors.toList());
+      return specificIds.toArray(new String[specificIds.size()]);
+    } catch (SQLException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)